protocols/bftcosi/bftcosi.go

```diff
@@ -97,23 +99,39 @@ type ProtocolBFTCoSi struct {
 
 	// finale signature that this BFTCoSi round has produced
 	finalSignature *BFTSignature
+
+	// our index in the Roster list
+	index int
+	// prepareSignature is the signature generated during the prepare phase
+	// This signature is adapted according to the exceptions that occured during
+	// the prepare phase.
+	prepareSignature []byte
+	// commitSignature is the signature generated during the commit phase
+	// This signature is adapted according to the exceptions that occured during
+	// the commit phase.
+	commitSignature []byte
 }
 
 // NewBFTCoSiProtocol returns a new bftcosi struct
 func NewBFTCoSiProtocol(n *sda.TreeNodeInstance, verify VerificationFunction) (*ProtocolBFTCoSi, error) {
 	// initialize the bftcosi node/protocol-instance
 	bft := &ProtocolBFTCoSi{
-		TreeNodeInstance: n,
-		suite:            n.Suite(),
-		prepare:          cosi.NewCosi(n.Suite(), n.Private()),
-		commit:           cosi.NewCosi(n.Suite(), n.Private()),
-		verifyChan:       make(chan bool),
-		doneProcessing:   make(chan bool, 2),
-		doneSigning:      make(chan bool, 1),
-		verificationFun:  verify,
-		AggregatedPublic: n.Roster().Aggregate,
-		threshold:        int(2.0 * math.Ceil(float64(len(n.Tree().List()))/3.0)),
-	}
+		TreeNodeInstance:     n,
+		suite:                n.Suite(),
+		prepare:              cosi.NewCosi(n.Suite(), n.Private(), n.Roster().Publics()),
+		commit:               cosi.NewCosi(n.Suite(), n.Private(), n.Roster().Publics()),
+		verifyChan:           make(chan bool),
+		doneProcessing:       make(chan bool, 2),
+		doneSigning:          make(chan bool, 1),
+		VerificationFunction: verify,
+		AggregatedPublic:     n.Roster().Aggregate,
+		threshold:            len(n.Tree().List()) * 2 / 3,
+		Msg:                  make([]byte, 0),
+		Data:                 make([]byte, 0),
+	}
+
+	idx, _ := n.Roster().Search(bft.ServerIdentity().ID)
```

Comment:
Well, I just came up with the first solution that came to my mind. 
It's a good question and unfortunatly, I can't answer yes to it because ... everything is exported :) So one can change the `roster.List` anyway we want. 
If this don't happen, we can still have many `ServerIdentity` in the array, so it will always return the first which may not be the right one, but at least it's consistent.